src/pgen/radiation_advection.cpp

```diff
@@ -103,7 +146,44 @@ void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
         }
       });
 
-  radiation::MomentPrim2Con(rc.get());
+  // Initialize samples
+  auto radpkg = pmb->packages.Get("radiation");
+  if (radpkg->Param<bool>("active")) {
+    if (radpkg->Param<std::string>("method") == "mocmc") {
+      radiation::MOCMCInitSamples(rc.get());
+    }
+  }
+
+  radiation::MomentPrim2Con(rc.get(), IndexDomain::interior);
+}
+
+void ProblemModifier(ParameterInput *pin) {
```

Comment:
At the moment I'm modifying some constants so `moment_*` always runs in scale-free mode and `mocmc` (and `monte_carlo`, though untested) always runs in scale mode. These can maybe be unified with an appropriately scale-free radiation distribution function inside `singularity-opac` in the (near?) future.